x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/LimitedRole.java

```diff
@@ -124,9 +124,7 @@ public IndicesAccessControl authorize(
      */
     @Override
     public IsResourceAuthorizedPredicate allowedIndicesMatcher(String action) {
-        IsResourceAuthorizedPredicate predicate = baseRole.indices().allowedIndicesMatcher(action);
```

Comment:
Fixing this to work with a `LimitedRole` nested beyond one level. The old implementation throws an `UnsupportedOperationException` since we call `indices()` on a `LimitedRole`. I'll create Jira task to track implementing this (and other methods in this class that have the same problem).